src/conn/conn_dhandle.c

```diff
@@ -314,8 +314,6 @@ __wt_conn_btree_open(
 	    F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE) &&
 	    !LF_ISSET(WT_DHANDLE_LOCK_ONLY));
 
-	WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_CLOSING));
```

Comment:
The reason I removed it was because it meant we could set `WT_CONN_CLOSING` before connection close started shutting down subsystems. In other words, I wanted `WT_CONN_CLOSING` to mean "we're closing", rather than, "we've reached a point in closing after which we'll no longer open files".

What kind of bugs are you thinking of? Why would it be an error to open a file after connection close has started?

We could, of course, add another flag `WT_CONN_CLOSING_NO_OPEN`, that would mean no files should be opened after this time.